### ISSUE TITLE
coverage ignore?

### ISSUE THREAD

**Issue by tunnckoCore** [hapijs/lab#329]
Any way to ignore some lines from coverage?

btw, awful codebase!

**Comment by AdriVanHoudt** [comment]
Not that I know of, you should always go for 100% coverage and otherwise you can lower the threshold. 
Also don't expect to much help if you are telling people their code base is awful!

**Comment by tunnckoCore** [comment]
Yea I see the `-t` option, but it seems that it not works.
```
lab test.js -cv -t 98
```

**Comment by gergoerdosi** [comment]
You can ignore lines by adding `$lab:coverage:off$` and `$lab:coverage:on$` around the code.

**Comment by tunnckoCore** [comment]
@gergoerdosi  without comments?

```js
  $lab:coverage:off$
  return got.get(api, opts).then(function _then(res) {
    var body = JSON.parse(res[0]);
    return body.token;
  });
```

**Comment by tunnckoCore** [comment]
doh.. right within comments.

awful.

**Comment by gergoerdosi** [comment]
No, within comments. Sorry, forgot to mention that.

    // $lab:coverage:off$
    return got.get(api, opts).then(function _then(res) {
        var body = JSON.parse(res[0]);
        return body.token;
    });
    // $lab:coverage:on$

**Comment by geek** [comment]
@tunnckoCore please explain what is awful.

**Comment by tunnckoCore** [comment]
@geek Code ordering and entire logic.

In my case as normal developer I decide to look over the docs and through code directly and follow it, before open an issue. Always when I should review some hapijs org repo/lib a freak out, lol. :-D But the ideas are good. I like lab, bossy, code and principles behind them, but I dont like the code behind them.

So.. I decide to look what lab doing when we provide `-c` as option.
It comes on [lib/coverage.js#L39](https://github.com/hapijs/lab/blob/master/lib/coverage.js#L39) after that moves to `internals.pattern` doing some `internals.escape` after that goes to [Transfrom.install#L23-L45](https://github.com/hapijs/lab/blob/master/lib/transform.js#L23-L45) passing it `options` and [`internals.prime` function from `coverage.js` #L23-L36](https://github.com/hapijs/lab/blob/master/lib/coverage.js#L23-L36). So I go to `lib/transform.js` and what I see.. doing some loop over `transform.js/internals.transofrms` and invokes `internals.prime` from `coverage.js`, which creating some messy function (require.extensions - which I dont see where is used) for every extension given in `transform.js internals.transfroms`

I mean.. doh.

**Comment by tunnckoCore** [comment]
I would be good to export the coverage tool.

**Comment by geek** [comment]
Please log an issue for the code flow problem you see in transforms and we can try to refactor it some.

**Comment by ldesplat** [comment]
require.extensions is a way to extend nodejs' way of require files. In the case of coverage support, we need to instrument the code and therefore inject code into each file before it is compiled. So, this is the basic case for coverage, but also Lab now supports transforming or transpiling your code through the `-T` option. So, to also support that use case with and without coverage support, we need to do a similar step to what happened in coverage & integrate with it. That's why the coverage module is now calling the transform.js file.

**Comment by jcollum-cambia** [comment]
from @tunnckoCore 's profile "Perfectionist"; ah, well there it is, no wonder he doesn't think your code is good enough

**Comment by tunnckoCore** [comment]
Haha, too old thread bump, but yea. Still. But that's not bad thing, everyone has habits, view and preferences. The only fact is that it's hard to follow and I assume to maintain, but this isn't totally correct too, cuz it's hard to me but not for these that write in that style. :) It is organization style, so they most are okey with it.
